pykechain/client.py

```diff
@@ -1341,6 +1343,137 @@ def create_service(self, name, scope, description=None, version=None,
 
         return service
 
+    def create_scope(self, name, status=None, description=None, tags=None, start_date=None, due_date=None,
+                     team=None):
+        """
+        Create a Scope.
+
+        :param name: Name of the scope
+        :type name: basestring
+        :param status: choose one of the :class:`enums.ScopeStatus`, defaults to `ScopeStatus.ACTIVE`
+        :type status: basestring or None
+        :param description: (optional) Description of the scope
+        :type description: basestring or None
+        :param tags: (optional) List of tags to be added to the new scope
+        :type tags: list or None
+        :param start_date: (optional) start date of the scope
+        :type start_date: datetime.datetime or None
+        :param due_date: (optional) due date of the scope
+        :type due_date: datetime.datetime or None
+        :param team: Team object
+        :type team :class: `models: Team`
+        :return: the created :class:`models.Scope`
+        :raises APIError: In case of failure of the creation of new Scope
+        """
+        assert isinstance(name, (str, text_type))
```

Comment:
assert is not robust enough when you [run python in optimised mode](https://github.com/emre/notes/blob/master/python/when-to-use-assert.md).

> if you run Python with the -O or -OO optimisation flags, and consequently there is no guarantee that assert statements will actually be run.



```suggestion
        if not isinstance(name, (str, text_type)):
            raise IllegalArgumentError(....)        
```